docs/api-reference/extensions/data-filter-extension.md

```diff
@@ -153,6 +154,61 @@ Format:
 * If `filterSize` is `1`: `[softMin, softMax]`
 * If `filterSize` is `2` to `4`: `[[softMin0, softMax0], [softMin1, softMax1], ...]` for each filtered property, respectively.
 
+##### `getFilterCategory` ([Function](../../developer-guide/using-layers.md#accessors), optional) {#getfiltercategory}
+
+* Default: `0`
+
+Called to retrieve the category for each object that it will be filtered by. Returns either a category as a number or string (if `categorySize: 1`) or an array.
+
+For example, consider data in the following format:
+
+```json
+[
+  {"industry": "retail", "coordinates": [-122.45, 37.78], "size": 10},
+  ...
+]
+```
+
+To filter by industry:
+
+```js
+new ScatterplotLayer({
+  data,
+  getPosition: d => d.coordinates,
+  getFilterCategory: d => d.industry,
+  filterCategories: ['retail', 'health'],
+  extensions: [new DataFilterExtension({categorySize: 1})]
+})
+```
+
+To filter by both industry and size:
+
+```js
+new ScatterplotLayer({
+  data,
+  getPosition: d => d.coordinates,
+  getFilterCategory: d => [d.industry, d.size],
+  filterCategories: [['retail', 'health'], [10, 20, 50]],
+  extensions: [new DataFilterExtension({categorySize: 2})]
+})
+```
+
+##### `filterCategories` (Array, optional) {#filtercategories}
+
+* Default: `[0]`
+
+The list of categories that should be rendered. If an object's filtered category is in the list, the object will be rendered; otherwise it will be hidden. This prop can be updated on user input or animation with very little cost.
+
+Format:
+
+* If `categorySize` is `1`: `['category1', 'category2']`
+* If `categorySize` is `2` to `4`: `[['category1', 'category2', ...], ['category3', ...], ...]` for each filtered property, respectively.
```

Comment:
Could we avoid the overloading and always required nested array, and just use firm typescript typing?

Reply:
Could it be a map with column names?

Reply:
[Again](https://github.com/visgl/deck.gl/pull/7915#discussion_r1497716672), I prioritized consistency with existing API here